roles/replace_node/tasks/main.yml

```diff
@@ -16,27 +33,37 @@
     glusterd_libdir: "/var/lib/glusterd"
     peer_tmp_dir: "{{ tmpdir['path'] }}"
 
-
+# Copies the authorized_keys file to the new node
 - import_tasks: authorization.yml
   when: gluster_maintenance_old_node is defined and
         gluster_maintenance_cluster_node is defined and
         gluster_maintenance_cluster_node_2 is defined
 
+# Peer restoration
 - import_tasks: peers.yml
   when: gluster_maintenance_old_node is defined and 
         gluster_maintenance_cluster_node is defined and
-        gluster_maintenance_cluster_node_2 is defined
+        gluster_maintenance_cluster_node_2 is defined 
 
+# volume restoration
 - import_tasks: volume.yml
   when: gluster_maintenance_old_node is defined and
         gluster_maintenance_cluster_node is defined and
         gluster_maintenance_cluster_node_2 is defined
 
+# Detach the old host, to replace host with different FQDN usecase
```

Comment:
shouldnt this task be executed before peer restoration?

Reply:
No. Because, there will be bricks in the volume that refers to the bricks in that host. So first replace-brick should be done first, followed by detach peer.